Automattic/pocket-casts-ios
# Tests: add a custom delegate for tests

Adds a custom delegate/screen when running the unit tests. This has two advantages:

1. It doesn't start the whole app, making running unit tests faster;
2. Given it doesn't start the app, you can debug without app interferences

An example of the number (2) is adding a breakpoint on `Theme.init`, for example. If you do that on this branch, you'll see that the breakpoint is hit because of unit tests. Do the same on `trunk` and you'll see that it hits because of the main app execution.

I also added a screen with some custom animation because why not.

https://user-images.githubusercontent.com/7040243/181255375-8d5154f1-e083-450d-ac2a-25a2f5cfeed9.mp4

## Additional information

Given we use Scene Delegates (because of CarPlay) I had to make use of a private API to achieve the desired result. This is not an issue because the private API call exists on the unit tests target only. The downside of that is that it can change at any time.

Basically, creating scenes works with some sort of caching, so in order to have the unit tests screen, we have to remove the scene sessions. If you're interesting in knowing more about that I recommend this article: https://hacknicity.medium.com/how-to-switch-your-ios-app-and-scene-delegates-for-improved-testing-9746279378c3

## To test

1. Run the app
2. ✅ Make sure the app runs without any issues
3. Run unit tests
4. ✅ Make sure the custom screen appears and unit tests run

## Checklist
